### ISSUE TITLE
Feature Request: Nested 'it' blocks should throw a useful error

### ISSUE THREAD

**Issue by juliemr** [jasmine/jasmine#1295]
Version 2.5.3 and previous, all environments.

Nested `it` blocks are unsupported and meaningless, but sometimes test writers mess up and make them anyway (for example, if they have helper functions). It would be nice if Jasmine threw an error when this happens, instead of silently passing. For example, something like:

```js
describe('nested its', function() {
  it('should throw some sort of error', function() {
    it('erroneous nested it', function() {
      expect(true).toEqual(false);
    });
  });
});
```

Should throw something like `unexpected "it" statement inside a test`.

**Comment by slackersoft** [comment]
This seems reasonable as we've moved to some nicer error messages for cases like this. I'd be happy to review a pull request to have `it` throw an error if Jasmine is already executing another `it`.

**Comment by Goodwine** [comment]
Note that the problem is not only nested `it`s, but also nested `beforeEach`, `afterEach`, etc.

**Comment by ksvitkovsky** [comment]
@slackersoft I started to work on the issue, should I keep the issue just about nested `it` statements or expand that to describes/befores/afters that are being used not on top-level of `describe` function?

**Comment by slackersoft** [comment]
To clarify, anything can be nested inside of a `describe` (including another `describe`), but nothing should be nested inside of an `it`, `beforeEach`, `afterEach`, `beforeAll`, or `afterAll`. Possibly the best way to block this is just to have all of these top-level interface methods throw an error if they are called while the suite is executing.

**Comment by sgravrock** [comment]
This was released in 2.9.0. Thanks, @ksvitkovsky!
